addons/l10n_ro/models/res_partner.py

```diff
@@ -18,3 +18,13 @@ def _commercial_fields(self):
         return super(ResPartner, self)._commercial_fields() + ['nrc']
 
     nrc = fields.Char(string='NRC', help='Registration number at the Registry of Commerce')
+
+    @api.depends('vat', 'country_id')
+    def _compute_company_registry(self):
+        # OVERRIDE
+        # In Romania, if you have a VAT number, it's also your company registry (CUI) number
+        super()._compute_company_registry()
+        for partner in self.filtered(lambda p: p.country_id.code == 'RO' and p.vat):
+            vat_country, vat_number = self._split_vat(partner.vat)
+            if vat_country == 'ro' and self.simple_vat_check(vat_country, vat_number):
```

Comment:
@antoine162 Please check https://github.com/odoo/odoo/pull/125058#issuecomment-1623836753

Reply:
Thanks. I didn't think about the case where the user puts the VAT number without the 'RO' prefix. Fixed now in the same way as Brice's PR.